### ISSUE TITLE
Java Error - java.io.IOException LongWritable

### ISSUE THREAD

**Issue by webtoed** [elastic/elasticsearch-hadoop#444]
Search nodes listed in es.nodes are the only nodes with HTTP.  They are my client nodes. 
I am trying to isolate this from my other issue(436). This was a test against another environment. 

The es.query works as expected  with _search post against the listed cluster (below hive statement). It also has worked in HIVE against my other Elastic cluster (which has HTTP enabled on data nodes). I have tried this without the size parameter,below. 

*Slightly off topic: es.query will not use , size=5 in the query. I was hoping to use this to restrict my data pull for testing. I can open another issue if this is not as expected.  
_________________________________
ERROR:
```
java.io.IOException: org.apache.hadoop.hive.ql.metadata.HiveException: java.lang.ClassCastException: org.apache.hadoop.io.LongWritable cannot be cast to org.apache.hadoop.io.Text
```
&
```
java.io.IOException: org.apache.hadoop.hive.ql.metadata.HiveException: java.lang.ClassCastException: org.apache.hadoop.io.LongWritable cannot be cast to org.apache.hadoop.io.Text
```
_________________________________
JAR: elasticsearch-hadoop-2.1.0.Beta4.jar 
ELASTIC_Version: 1.1.1 
_________________________________
HIVE_Command:
```
CREATE EXTERNAL TABLE dc01_errors_mobile (
  date timestamp,
  category string,
  subcategory string,
  application string,
  version string,
  company_id string,
  entity_code string,
  user_id string,
  description string)
STORED BY 'org.elasticsearch.hadoop.hive.EsStorageHandler'
TBLPROPERTIES('es.resource'= 'errors-2015.04.30/errors',
        'es.nodes' = '10.1.6.203,10.1.6.204,10.1.6.205',
        'es.nodes.client.only' = 'true',
        'es.field.read.empty.as.null' = 'true',
        'es.mapping.id' = '_id',
        'es.query'='{"query":{"query_string":{"query":"application:MobileMetric AND @timestamp:[now-5m TO now]"}},"size":5}',
        'es.mapping.names' = 'date:@timestamp,
                              category:category, 
                              subcategory:subcategory, 
                              application:application, 
                              version:version,
                              company_id:company_id,
                              entity_code:entity_code,
                              user_id:user_id,
                              description:description,
                              entity_code:entity_code');          
```        
-- Stream Data
SELECT * FROM dc01_errors_mobile;
_________________________________
LOG:
```
15/04/30 14:42:01 INFO log.PerfLogger: <PERFLOG method=compile from=org.apache.hadoop.hive.ql.Driver>
15/04/30 14:42:01 INFO log.PerfLogger: <PERFLOG method=parse from=org.apache.hadoop.hive.ql.Driver>
15/04/30 14:42:01 INFO parse.ParseDriver: Parsing command: SELECT * FROM `user01.dc01r_errors_mobile`
15/04/30 14:42:01 INFO parse.ParseDriver: Parse Completed
15/04/30 14:42:01 INFO log.PerfLogger: </PERFLOG method=parse start=1430430121134 end=1430430121134 duration=0 from=org.apache.hadoop.hive.ql.Driver>
15/04/30 14:42:01 INFO log.PerfLogger: <PERFLOG method=semanticAnalyze from=org.apache.hadoop.hive.ql.Driver>
15/04/30 14:42:01 INFO parse.SemanticAnalyzer: Starting Semantic Analysis
15/04/30 14:42:01 INFO parse.SemanticAnalyzer: Completed phase 1 of Semantic Analysis
15/04/30 14:42:01 INFO parse.SemanticAnalyzer: Get metadata for source tables
15/04/30 14:42:01 INFO parse.SemanticAnalyzer: Get metadata for subqueries
15/04/30 14:42:01 INFO parse.SemanticAnalyzer: Get metadata for destination tables
15/04/30 14:42:01 INFO ql.Context: New scratch dir is hdfs://dc01r1mgmthdp01.companyasp.com:8020/tmp/hive-hive/hive_2015-04-30_14-42-01_134_8904756159309137145-377
15/04/30 14:42:01 INFO parse.SemanticAnalyzer: Completed getting MetaData in Semantic Analysis
15/04/30 14:42:01 INFO common.FileUtils: Creating directory if it doesn't exist: hdfs://dc01r1mgmthdp01.companyasp.com:8020/tmp/hive-hive/hive_2015-04-30_14-42-01_134_8904756159309137145-377/-mr-10000/.hive-staging_hive_2015-04-30_14-42-01_134_8904756159309137145-377
15/04/30 14:42:01 INFO parse.SemanticAnalyzer: Set stats collection dir : hdfs://dc01r1mgmthdp01.companyasp.com:8020/tmp/hive-hive/hive_2015-04-30_14-42-01_134_8904756159309137145-377/-mr-10000/.hive-staging_hive_2015-04-30_14-42-01_134_8904756159309137145-377/-ext-10002
15/04/30 14:42:01 INFO ppd.OpProcFactory: Processing for FS(651710)
15/04/30 14:42:01 INFO ppd.OpProcFactory: Processing for SEL(651709)
15/04/30 14:42:01 INFO ppd.OpProcFactory: Processing for TS(651708)
15/04/30 14:42:01 INFO parse.SemanticAnalyzer: Completed plan generation
15/04/30 14:42:01 INFO ql.Driver: Semantic Analysis Completed
15/04/30 14:42:01 INFO log.PerfLogger: </PERFLOG method=semanticAnalyze start=1430430121134 end=1430430121178 duration=44 from=org.apache.hadoop.hive.ql.Driver>
15/04/30 14:42:01 INFO exec.TableScanOperator: Initializing Self 651708 TS
15/04/30 14:42:01 INFO exec.TableScanOperator: Operator 651708 TS initialized
15/04/30 14:42:01 INFO exec.TableScanOperator: Initializing children of 651708 TS
15/04/30 14:42:01 INFO exec.SelectOperator: Initializing child 651709 SEL
15/04/30 14:42:01 INFO exec.SelectOperator: Initializing Self 651709 SEL
15/04/30 14:42:01 INFO exec.SelectOperator: SELECT struct<date:timestamp,category:string,subcategory:string,application:string,version:string,company_id:string,entity_code:string,user_id:string,description:string>
15/04/30 14:42:01 INFO exec.SelectOperator: Operator 651709 SEL initialized
15/04/30 14:42:01 INFO exec.SelectOperator: Initializing children of 651709 SEL
15/04/30 14:42:01 INFO exec.ListSinkOperator: Initializing child 651711 OP
15/04/30 14:42:01 INFO exec.ListSinkOperator: Initializing Self 651711 OP
15/04/30 14:42:01 INFO exec.ListSinkOperator: Operator 651711 OP initialized
15/04/30 14:42:01 INFO exec.ListSinkOperator: Initialization Done 651711 OP
15/04/30 14:42:01 INFO exec.SelectOperator: Initialization Done 651709 SEL
15/04/30 14:42:01 INFO exec.TableScanOperator: Initialization Done 651708 TS
15/04/30 14:42:01 INFO ql.Driver: Returning Hive schema: Schema(fieldSchemas:[FieldSchema(name:user01.dc01r_errors_mobile.date, type:timestamp, comment:null), FieldSchema(name:user01.dc01r_errors_mobile.category, type:string, comment:null), FieldSchema(name:user01.dc01r_errors_mobile.subcategory, type:string, comment:null), FieldSchema(name:user01.dc01r_errors_mobile.application, type:string, comment:null), FieldSchema(name:user01.dc01r_errors_mobile.version, type:string, comment:null), FieldSchema(name:user01.dc01r_errors_mobile.company_id, type:string, comment:null), FieldSchema(name:user01.dc01r_errors_mobile.entity_code, type:string, comment:null), FieldSchema(name:user01.dc01r_errors_mobile.user_id, type:string, comment:null), FieldSchema(name:user01.dc01r_errors_mobile.description, type:string, comment:null)], properties:null)
15/04/30 14:42:01 INFO log.PerfLogger: </PERFLOG method=compile start=1430430121134 end=1430430121179 duration=45 from=org.apache.hadoop.hive.ql.Driver>
15/04/30 14:42:01 INFO log.PerfLogger: <PERFLOG method=Driver.run from=org.apache.hadoop.hive.ql.Driver>
15/04/30 14:42:01 INFO log.PerfLogger: <PERFLOG method=TimeToSubmit from=org.apache.hadoop.hive.ql.Driver>
15/04/30 14:42:01 INFO log.PerfLogger: <PERFLOG method=acquireReadWriteLocks from=org.apache.hadoop.hive.ql.Driver>
15/04/30 14:42:01 INFO lockmgr.DummyTxnManager: Creating lock manager of type org.apache.hadoop.hive.ql.lockmgr.zookeeper.ZooKeeperHiveLockManager
15/04/30 14:42:01 INFO zookeeper.ZooKeeper: Initiating client connection, connectString=dc01r1mgmthdp03.companyasp.com:2181,dc01r1mgmthdp01.companyasp.com:2181,dc01r1mgmthdp02.companyasp.com:2181 sessionTimeout=600000 watcher=org.apache.hadoop.hive.ql.lockmgr.zookeeper.ZooKeeperHiveLockManager$DummyWatcher@1f6a45b
15/04/30 14:42:01 INFO log.PerfLogger: </PERFLOG method=acquireReadWriteLocks start=1430430121182 end=1430430121195 duration=13 from=org.apache.hadoop.hive.ql.Driver>
15/04/30 14:42:01 INFO log.PerfLogger: <PERFLOG method=Driver.execute from=org.apache.hadoop.hive.ql.Driver>
15/04/30 14:42:01 INFO ql.Driver: Starting command: SELECT * FROM `user01.dc01r_errors_mobile`
15/04/30 14:42:01 INFO log.PerfLogger: </PERFLOG method=TimeToSubmit start=1430430121181 end=1430430121196 duration=15 from=org.apache.hadoop.hive.ql.Driver>
15/04/30 14:42:01 INFO log.PerfLogger: <PERFLOG method=runTasks from=org.apache.hadoop.hive.ql.Driver>
15/04/30 14:42:01 INFO log.PerfLogger: </PERFLOG method=runTasks start=1430430121196 end=1430430121196 duration=0 from=org.apache.hadoop.hive.ql.Driver>
15/04/30 14:42:01 INFO log.PerfLogger: <PERFLOG method=PostHook.com.cloudera.navigator.audit.hive.HiveExecHookContext from=org.apache.hadoop.hive.ql.Driver>
15/04/30 14:42:01 INFO log.PerfLogger: </PERFLOG method=PostHook.com.cloudera.navigator.audit.hive.HiveExecHookContext start=1430430121196 end=1430430121197 duration=1 from=org.apache.hadoop.hive.ql.Driver>
15/04/30 14:42:01 INFO log.PerfLogger: </PERFLOG method=Driver.execute start=1430430121196 end=1430430121197 duration=1 from=org.apache.hadoop.hive.ql.Driver>
15/04/30 14:42:01 INFO ql.Driver: OK
15/04/30 14:42:01 INFO log.PerfLogger: <PERFLOG method=releaseLocks from=org.apache.hadoop.hive.ql.Driver>
15/04/30 14:42:01 INFO ZooKeeperHiveLockManager:  about to release lock for user01/dc01r_errors_mobile
15/04/30 14:42:01 INFO ZooKeeperHiveLockManager:  about to release lock for user01
15/04/30 14:42:01 INFO log.PerfLogger: </PERFLOG method=releaseLocks start=1430430121197 end=1430430121204 duration=7 from=org.apache.hadoop.hive.ql.Driver>
15/04/30 14:42:01 INFO log.PerfLogger: </PERFLOG method=Driver.run start=1430430121181 end=1430430121204 duration=23 from=org.apache.hadoop.hive.ql.Driver>
```
_________________________________

**Comment by webtoed** [comment]
@costin what does a label 'ready' mean?

**Comment by costin** [comment]
@webtoed it's a label applied by waffle.io for issues in the pipeline, _ready_ to be looked at. Not the best choice, I agree.

**Comment by webtoed** [comment]
@costin Any thoughts with this issue?

**Comment by webtoed** [comment]
Thoughts?

**Comment by costin** [comment]
@webtoed Sorry for the silence - I'm on PTO and will update once I return.
Thank you for understanding.

**Comment by webtoed** [comment]
when will you be back from PTO?

**Comment by costin** [comment]
@webtoed I'm back now.
I have added minor formatting to your initial post to make it more readable. 

First off, the exception is caused by what looks like a type mismatch between your index mapping and your Hive table. That is, the connector reads the data from Elasticsearch and creates a `LongWritable` (that's because that is mapped in the index as a long) but in the table it is created as a `String`/`Text`.

What is your index mapping?

Further more, you don't need to define the mapping for every field - if the name in ES and Hive are the same, the connector will use that automatically.
As for `size` - note that is applied per shard as explained [here](https://www.elastic.co/guide/en/elasticsearch/guide/master/scan-scroll.html).

**Comment by costin** [comment]
One last comment, I _strongly_ suggest upgrading Elasticsearch to 1.5.x or the soon to be released 1.6
There are a LOT of improvements done both in terms of efficiency and reliability.
